src/Common/src/System/Security/Cryptography/Pkcs12Kdf.cs

```diff
@@ -107,7 +107,15 @@ internal static class Pkcs12Kdf
             int SLen = ((salt.Length - 1 + vBytes) / vBytes) * vBytes;
 
             // The password is a null-terminated UTF-16BE version of the input.
-            int passLen = (password.Length + 1) * 2;
+            int passLen = checked((password.Length + 1) * 2);
+
+            // "" will have a pointer value pointing at the terminating \0, and length 0.
+            // null will have a pointer pointing to null and length 0.
```

Comment:
Those aren't the only two options, of course.  `"foo".AsSpan(1, 0)` will have a ref/pointer value that's not null and a length 0, but it won't be pointing at `\0`.